main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+var resultVictory int
+var resultDefeat int
+
+func main() {
+	var wantToPlay string
+	fmt.Println("Bem vindo ao Cara ou Coroa! Deseja jogar [sim/nao]?")
+	fmt.Scan(&wantToPlay)
+
+	for {
+		if strings.ToLower(wantToPlay) == "sim" || strings.ToLower(wantToPlay) == "s" {
+			play()
+		} else if strings.ToLower(wantToPlay) == "nao" || strings.ToLower(wantToPlay) == "n" {
+			fmt.Println("Obrigado.")
+			break
+		}
+		main()
+		break
+	}
+}
+
+func play() {
+	var userCurrency string
+
+	println("Escolha cara ou coroa.")
+	fmt.Scan(&userCurrency)
+
+	if strings.ToLower(userCurrency) == "cara" || strings.ToLower(userCurrency) == "coroa" {
+		flipACoin(userCurrency)
+	} else {
+		fmt.Println("Por favor digite cara ou coroa.")
+		play()
+	}
+}
+
+func flipACoin(userCurrency string) {
+
+	coin := []string{
+		"cara",
+		"coroa",
+	}
+	rand.Seed(time.Now().UnixNano())
+	side := coin[rand.Intn(len(coin))]
+	if strings.ToLower(userCurrency) == side {
+		fmt.Println(side, "Você Ganhou!")
+		resultVictory++
+		playAgain()
+
+	} else {
+		fmt.Println(side, "Você Perdeu!")
+		resultDefeat++
+		playAgain()
+	}
+}
+
+func playAgain() {
```

Comment:
Você pode ignorar meu comentário anterior caso você for desfazer dessa função. Eu acredito que ela pode ficar dentro da função play(), sem problemas

Reply:
Feito!